### ISSUE TITLE
An interface to assigning of bundles to users with logic

### ISSUE THREAD

**Issue by FredrikKarlssonSpeech** [humlab-speech/visible-speech-deployment#136]
The user creation and bundle assignment interface now allow assigning the same bundle to multiple users, which creates a risk of users overwriting other's data. And, the interface currently does not take in "Done" flags in the interface and writes entirely new JSON files, which makes it impossible to know if a bundle is actually done or not once the project has received a new recording and the project leader assigns it to a person (when entirely new JSON files will be written). And we need a mechanism for self-assignement of "free" bundles.

We propose the following logic:

1) All not assigned and "Done" bundles belong to the project leader when the JSONs are read in. For all other members, the bundles are inactive (see next sentence). When assigned to a project member the "Done" flag is cleared and the bundle checkbox is deactivated (and the text put in italics?) for all other members including the project leader. 

2) The user may self-assign bundles in their list that are still active. That means that it will be presented as deactivated (taken) for anyone opening the pane later.

3) The user that currently "owns" the bundle sees a checkbox next to the bundle when it is marked as "Done" in the segment list it was read in from.

**Comment by FredrikKarlssonSpeech** [comment]
Risk cases:

* Adding a new recording.
* Self-assignment creates the possibility of multiple assignments of  a bundle - and creates the possibility of bundle annotations being overwritten.

**Comment by FredrikKarlssonSpeech** [comment]
I have looked at what can go wrong, and it does seem like concurrent editing of the same transcription will not block commits. Of course, one writer will overwrite the just a

https://user-images.githubusercontent.com/200246/171693500-bffa7f8f-3e10-4995-b400-f01e5f3a5bb4.mov

dded previous commit by a different user, but that is about the extent of the loss one might face in transcriptions.

But, the DONE flag in the bundle list will be overwritten / lost when multiple users are working on the same file of course.

**Comment by TomasSkotare** [comment]
So, if I understand the issue correctly, there are 3 issues we wish to avoid.

1) More than one person can be assigned the same bundle, leading to data loss if both people work simultaneously.

2) Each time the list in the user interface is modified, any "done" flag is cleared.

3) The web interface does not show whether or not a bundle has been marked as done by any user

To solve 1), the user interface should ensure that at most one user is assigned to any single bundle.

As for 2) and 3), the web interface will currently revert the bundle list of each user to the default state without taking the current "done" status into consideration. To solve 2) and 3), the web interface logic must consider the current status of the bundle lists, both to present this status to the user in the web interface and to ensure that "done" flags are not reset.

**Comment by FredrikKarlssonSpeech** [comment]
> To solve 2) and 3), the web interface logic must consider the current status of the bundle lists, both to present this status to the user in the web interface and to ensure that "done" flags are not reset.

Yes, the flag needs to be read from the JSON, but if the bundle is marked as Done by the user it has been assigned to, it should be reassigned back to the project leader in the interface.
